### ISSUE TITLE
Structure of the MI-GAN inference model different from the one used for training?

### ISSUE THREAD

**Issue by anirao26** [Picsart-AI-Research/MI-GAN#10]
Hello,

Thank you for the amazing and insightful work! I was trying to train the model on my own set of images. However, I noticed that the number of parameters in the MIGAN generator part of the model that is used to train the model is very different from the one for which the checkpoint was provided (52 M vs 5.9 M respectively). Maybe I missed something here but could you please help me understand this? Thanks!

**Comment by AndranikSargsyan** [comment]
Hello. During training the model size is bigger, because our method uses a model reparameterization trick as described in the section 3.3 of the paper. Once the model training is finished you need to fuse the weights in order to decrease the model size for inference. For fusion, you can use the [`scripts/export_inference_model.py`](https://github.com/Picsart-AI-Research/MI-GAN/blob/main/scripts/export_inference_model.py) script. You can find the details of how to use that script at the end of [README's Training section](https://github.com/Picsart-AI-Research/MI-GAN?tab=readme-ov-file#training).

**Comment by anirao26** [comment]
Thank you so much for the clarification. I wanted to fine tune the model instead of training the whole model again. The layers don't match exactly so I was wondering the best way to do it?

**Comment by AndranikSargsyan** [comment]
Training checkpoints are provided [in this Google Drive folder](https://drive.google.com/drive/folders/1NneiN4LcVy_YxJaN0S2cYDoVN9ujahBU).  You can download the corresponding (unfused) training checkpoint from there and use it for fine-tuning.

**Comment by anirao26** [comment]
Thank you @AndranikSargsyan! One more question, in the code, can you tell me what the **G_ema** variable signifies? I thought the **G** variable was the MI-GAN generator model and **D** is the discriminator.

**Comment by AndranikSargsyan** [comment]
`G_ema` keeps track of exponential moving average (ema) of the generator weights during training. Exponential moving averaging of the generator weights is a popular strategy for improving generator performance.

**Comment by anirao26** [comment]
Thank you so much for your quick response @AndranikSargsyan. One final thing I wanted to ask you was about the training phase. In the code, I see that there are two instances of G (Gmain and GReg) and D (Dmain and DReg) since in the default setting regularization interval is set to 4 and 16 respective;y. I am unable to understand why we have two instances of the generator and discriminator models in this case. Really appreciate your help!

**Comment by AndranikSargsyan** [comment]
There're actually no two instances of generator and discriminator (except the G_ema for generator). Gmain/Greg and Dmain/Dreg are just strings which signify the current phase of the training: i.e. Gmain/Dmain signify that there should be no regularization in the loss computation of current iteration, while Greg/Dreg signify that loss computation should include regularization component such as R1 used in the training of the discriminator of MI-GAN.
